plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/polaris/PolarisSchema.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.metastore.polaris;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents a table schema from Polaris.
+ * This is used for both Iceberg and Generic tables.
+ */
+public class PolarisSchema
```

Comment:
Try using simple Java records for this.  They automatically support Jackson encoding and are much simpler to write.  Looks like Polaris is using Kebab case so you will need:
```
@JsonNaming(KebabCaseStrategy.class)